### PR TITLE
Dockerize Chatbot

### PR DESCRIPTION
**Pull request by markediez**
Dockerfile is made for chatbot

Create the image by running
docker build -t <name>:<tag> <path> 

Run using
docker run <name>:<tag>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM ruby:2.3
+
+# Install necessary libraries
+RUN apt-get update
```
Comment (cthielen) [inline Dockerfile]: I'm not familiar enough with Docker to know the answer to this: is it necessary to run updates or is that the upstream image's responsibility?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+WORKDIR $HOME
+ADD Gemfile* $HOME/
+
+# Install gems
+RUN gem install ruby-ldap -v '0.9.17'
+RUN bundle install
+ADD . $HOME
+
+# Run chatbot
+CMD ["./dss-chatbot.rb", "run"]
```
Comment (cthielen) [inline Dockerfile]: You'll need to modify the logger to log to STDOUT and STDERR instead of a file to be properly Dockerized.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+RUN apt-get install -y libldap2-dev
+
+# Set up where ruby app will be stationed
+# https://www.packet.net/blog/how-to-run-your-rails-app-on-docker/
+ENV HOME /home/chatbot
+WORKDIR $HOME
+ADD Gemfile* $HOME/
+
+# Install gems
+RUN gem install ruby-ldap -v '0.9.17'
```
Comment (cthielen) [inline Dockerfile]: Shouldn't this be "bundle install" instead?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+# Install necessary libraries
+RUN apt-get update
+RUN apt-get install libsasl2-dev
+RUN apt-get install libssl-dev
+RUN apt-get install -y libldap2-dev
+
+# Set up where ruby app will be stationed
+# https://www.packet.net/blog/how-to-run-your-rails-app-on-docker/
+ENV HOME /home/chatbot
```
Comment (cthielen) [inline Dockerfile]: Is this necessary?

File: Dockerfile (hunk 2 above)
Comment (cthielen) [inline Dockerfile]: Also, where is settings.yml copied in?

File: Dockerfile (hunk 3 above)
Comment (markediez) [inline Dockerfile]: It was, but it outputted an error telling me to run that command first prior to bundle install

File: Dockerfile (hunk 1 above)
Comment (markediez) [inline Dockerfile]: I'm not sure if its the upstream's responsibility however I do know I need to run it to install the other libraries

File: Dockerfile (hunk 3 above)
Comment (cthielen) [inline Dockerfile]: What's the error?

File: Dockerfile (hunk 4 above)
Comment (markediez) [inline Dockerfile]: Nope

File: Dockerfile (hunk 2 above)
Comment (markediez) [inline Dockerfile]: In this code it would be handled in line 18. So $HOME/config/settings.yml

Comment (cthielen) [inline Dockerfile]: But where does settings.yml come from? WORKDIR simply sets the CWD within the container.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ADD ./commands/help.rb $HOME/commands
+ADD ./commands/host.rb $HOME/commands
+ADD ./commands/iam.rb $HOME/commands
+ADD ./commands/ldap.rb $HOME/commands
+ADD ./commands/mail.rb $HOME/commands
+ADD ./commands/roles.rb $HOME/commands
+ADD ./commands/visioneers.rb $HOME/commands
+
+# Update crontab from whenever gem
+RUN whenever --update-crontab
```
Comment (cthielen) [inline Dockerfile]: cron won't work anymore. What was it used for?

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+RUN apt-get install libssl-dev
+RUN apt-get install -y libldap2-dev
+RUN apt-get install -y gnome-schedule
+
+# Set up environment to avoid adding files to root home
+ENV HOME /home
+WORKDIR $HOME
+
+# Install gems
+ADD Gemfile* $HOME/
```
Comment (cthielen) [inline Dockerfile]: Change this to ADD Gemfile and Gemfile.lock, not Gemfile*

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,44 @@
+FROM ruby:2.3
+
+# Install necessary libraries
+RUN apt-get update
+RUN apt-get install libsasl2-dev
+RUN apt-get install libssl-dev
+RUN apt-get install -y libldap2-dev
+RUN apt-get install -y gnome-schedule
```
Comment (cthielen) [inline Dockerfile]: What's gnome-schedule for?

File: Dockerfile (hunk 5 above)
Comment (markediez) [inline Dockerfile]: run !assignments at 9 and 3(?)

File: Dockerfile (hunk 7 above)
Comment (markediez) [inline Dockerfile]: Cron task, it isn't built in

Comment (cthielen) [inline Dockerfile]: Where'd you get this from? Wouldn't installing gnome-schedule bring in all of X11, etc.?

Comment (markediez) [inline Dockerfile]: https://help.ubuntu.com/community/CronHowto

**Pull request merged**